src/Functions/parseTimeDelta.cpp

```diff
@@ -0,0 +1,243 @@
+                                    + " find number after '.' failed: " + String(str) + ".",
+                                ErrorCodes::BAD_ARGUMENTS);
+                        }
+                    }
+
+                    /// convert float/integer string to float
+                    Float64 base = 0;
+                    std::string_view base_str = str.substr(slow, fast - slow);
+                    auto value = boost::convert<Float64>(base_str, boost::cnv::strtol());
+                    if (!value.has_value())
+                    {
+                        throw Exception(
+                            "bad auguments invalid expression for function " + getName()
+                                + " convert string to float64 failed:  " + String(base_str) + ".",
+                            ErrorCodes::BAD_ARGUMENTS);
+                    }
+                    base = value.get();
+
+                    ScanIgnoreCharacters(str, fast, last_pos);
+                    slow = fast;
+
+                    /// scan a unit
+                    if (!ScanUnit(str, fast, last_pos))
+                    {
+                        throw Exception(
+                            "bad auguments invalid expression for function " + getName() + " find unit failed: " + String(str) + ".",
+                            ErrorCodes::BAD_ARGUMENTS);
+                    }
+
+                    /// get unit number
+                    std::string_view unit = str.substr(slow, fast - slow);
+                    auto iter = time_unit_to_float.find(unit);
+                    if (iter == time_unit_to_float.end()) /// not find unit
+                    {
+                        throw Exception(
+                            "bad auguments invalid expression for function " + getName() + " parse unit failed: " + String(iter->first)
+                                + ".",
+                            ErrorCodes::BAD_ARGUMENTS);
+                    }
+                    result += base * iter->second;
+                }
+
+                res_data.emplace_back(result);
+            }
+
+            return col_to;
+        }
+
+        /// scan an unsigned integer number
+        static bool ScanUnsignedInteger(std::string_view & str, Int64 & index, Int64 last_pos)
+        {
+            int64_t begin_index = index;
+            while (index <= last_pos && isdigit(str[index]))
+            {
+                index++;
+            }
+            return index != begin_index;
+        }
+
+        /// scan a unit
+        static bool ScanUnit(std::string_view & str, Int64 & index, Int64 last_pos)
+        {
+            int64_t begin_index = index;
+            while (index <= last_pos && isalpha(str[index]))
+            {
+                index++;
+            }
+            return index != begin_index;
+        }
+
+        /// scan for characters to ignore
+        static bool ScanIgnoreCharacters(std::string_view & str, Int64 & index, Int64 last_pos)
```

Comment:
This method returns bool but we don't use return value enywhere, so let's make it void
```suggestion
        static void ScanIgnoreCharacters(std::string_view & str, Int64 & index, Int64 last_pos)
```